once_test.py

```diff
@@ -729,6 +823,29 @@ async def counting_fn2():
         self.assertEqual(await counting_fn2(), 2)
         self.assertEqual(await counting_fn2(), 2)
 
+    async def test_once_per_thread(self):
+        counter = Counter()
+
+        @once.once(per_thread=True)
+        async def counting_fn(*args) -> int:
+            """Returns the call count, which should always be 1."""
```

Comment:
This one should have a barrier to test for parallel execution as well

Reply:
this code mixes sync and threading in an unpleasant way, so I was having trouble with getting this to work - using an async barrier resulted in lots of `is bound to a different event loop` eerrors.

Reply:
we did it - go pair programming